barrels/LogMonkey/Logger.mc

```diff
@@ -0,0 +1,172 @@
+using Toybox.Time.Gregorian as Greg;
+using Toybox.Lang as Lang;
+
+module LogMonkey {
+
+    //! The Logger class provides public APIs to log various types of
+    //! messages. Each Logger class is associated with a log level which
+    //! is set within the Logger.initialize() function.
+    class Logger {
+
+        private var mLogLevel, mLogStream;
+
+        //! Creates a new Logger object.
+        //! @param [Toybox::Lang::String] The log level value to log messages through this class to
+        //! @param [Toybox::Lang::Object] An object which defines a println(message) function
+        (:debug)
+        function initialize(logLevel, logStream) {
+            mLogLevel = logLevel;
+            mLogStream = logStream;
+        }
+
+        // An empty version of all functions in this class annotated
+        // with the :release tag are included to save memory when an
+        // app is compiled in release mode. We will still have the same
+        // symbols available in the release build so we won't get symbol
+        // not found errors while saving memory.
+        (:release)
+        function initialize(logLevel, logStream) {
+        }
+
+        // The string formats to use when printing log messages
+        private static var FORMAT_VARIABLE = "$1$=($2$) $3$";
+        private static var FORMAT_LOG_MESSAGE = "(lmf1)[$1$] {$2$} $3$: $4$";
+        private static var FORMAT_TIMESTAMP = "$1$-$2$-$3$ $4$:$5$:$6$";
```

Comment:
These private static are not necessary in release. They could easily be annotated out for additional savings.

Reply:
These should be taken care of with the empty class approach discussed above.